### PR TITLE
lib: nrf_modem: add POLLNVAL to new poll() offloading

### PR DESCRIPTION
**Pull request by lemrey**
POLLNVAL was not offloaded in the new implementation.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

**Review by eivindj-nordic — APPROVED** [review]

**Comment by lemrey** [comment]
Wait for @tokangas approval before merging.

**Comment by tokangas** [comment]
The fix doesn't seem to work. `poll()` now returns, but it returns `0` and `POLLNVAL` isn't raised. Helped @lemrey so he can reproduce the issue himself and can debug it quicker. Here's the JIRA item for this bug: NCSDK-19889

File: lib/nrf_modem_lib/nrf91_sockets.c
Comment (lemrey) [inline lib/nrf_modem_lib/nrf91_sockets.c]: ah, this should have been `flags`

File: lib/nrf_modem_lib/nrf91_sockets.c
Comment (lemrey) [inline lib/nrf_modem_lib/nrf91_sockets.c:815]: this can't be signaled at this point, so I removed this check and zero `flag` instead

File: lib/nrf_modem_lib/nrf91_sockets.c
Comment (lemrey) [inline lib/nrf_modem_lib/nrf91_sockets.c:821]: have removed these various calls to `k_poll_signal_raise`, now I raise once at the end of the function

**Review by tokangas — APPROVED** [review]
Verified with `modem_shell` that it works now, `POLLNVAL` is raised when the socket is closed.

**Pull request merged**
